imagebuildah/stage_executor.go

```diff
@@ -1246,6 +1287,18 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 			if err != nil {
 				return "", nil, fmt.Errorf("error committing container for step %+v: %w", *step, err)
 			}
+			// Following step is just built and was not used from cache so
+			// check if --cache-to was specified if yes then attempt pushing
+			// this cache to remote repo and fail accordingly
+			if cacheKey != "" && cacheID == "" && s.executor.cacheTo != "" {
+				// try to push this cache to remote repository
+				destSpec := fmt.Sprintf("%s:%s", s.executor.cacheTo, cacheKey)
+				logCachePush(destSpec)
+				err = s.pushCache(ctx, imgID, destSpec)
+				if err != nil {
```

Comment:
We can go ahead and merge these two lines.  We checked that `err` was `nil` before we got here, and if `pushCache()` returns an error, we return immediately.